infoschema/cluster.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infoschema
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/domain/infosync"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util"
+)
+
+// Cluster table list, attention:
+// 1. the table name should be upper case.
+// 2. clusterTableName should equal to "TIDB_CLUSTER_" + memTableTableName.
+const (
+	clusterTableSlowLog     = "TIDB_CLUSTER_SLOW_QUERY"
+	clusterTableProcesslist = "TIDB_CLUSTER_PROCESSLIST"
+)
+
+// memTableToClusterTableMap means add memory table to cluster table.
+var memTableToClusterTableMap = map[string]string{
+	tableSlowLog:     clusterTableSlowLog,
+	tableProcesslist: clusterTableProcesslist,
+}
+
+// clusterTableMap is the cluster table map.
+// It will be initialized in `init` function.
+var clusterTableMap = map[string]struct{}{}
+
+func init() {
+	var addrCol = columnInfo{"ADDRESS", mysql.TypeVarchar, 64, 0, nil, nil}
+	for memTableName, clusterMemTableName := range memTableToClusterTableMap {
+		memTableCols := tableNameToColumns[memTableName]
+		if len(memTableCols) == 0 {
+			continue
+		}
+		cols := make([]columnInfo, 0, len(memTableCols)+1)
+		cols = append(cols, memTableCols...)
+		cols = append(cols, addrCol)
+		tableNameToColumns[clusterMemTableName] = cols
+		clusterTableMap[clusterMemTableName] = struct{}{}
+	}
+}
+
+// isClusterTableByName used to check whether the table is a cluster memory table.
+func isClusterTableByName(dbName, tableName string) bool {
+	dbName = strings.ToLower(dbName)
+	switch dbName {
+	case util.InformationSchemaLowerName, util.PerformanceSchemaLowerName:
```

Comment:
It’s not a good idea to define these constants in subpackage `until`

Reply:
I define this before, Since those name will used in multiple pkg, and alway use a `const string` is bad, but define in `information_schema` will have circle import.